### PR TITLE
Modify rule S4624: Adapt to LaYC

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S4624/javascript/rule.adoc [hunk 1]
```diff
@@ -1,34 +1,32 @@
 == Why is this an issue?
 
-Template literals (previously named "template strings") are an elegant way to build a string without using the ``+`` operator to make strings concatenation more readable. 
+Template literals, also known as template strings, allow for string interpolation and multiline strings in JavaScript. They provide a more convenient and flexible way to work with strings compared to traditional string concatenation or manipulation.
 
-However, it's possible to build complex string literals by nesting together multiple template literals, and therefore lose readability and maintainability.
+Template literals are a convenient way to include variables or expressions within a string by using the backtick ``++(`)++`` character. They allow you to embed placeholders ``++`${expression}`++`` in the string and evaluate them dynamically.
```
Comment (victor-diez-sonarsource) [inline rules/S4624/javascript/rule.adoc:5]:
```suggestion
Template literals are delimited with the backtick ``++(`)++`` character. They are a convenient way to include variables or expressions within a string using placeholders ``++`${expression}`++`` in the string and evaluate them dynamically.
```

Comment (victor-diez-sonarsource) [reply rules/S4624/javascript/rule.adoc:5]: in the first phrase seems you imply that you can use the backtick to include variables. I would separate the backtick syntax from the placeholder one.

**Review by victor-diez-sonarsource — CHANGES_REQUESTED** [review]
Can you please mention this exception? https://github.com/SonarSource/eslint-plugin-sonarjs/issues/319

**Review by victor-diez-sonarsource — APPROVED** [review]
Please add also the reason for the exception. Originally requested in https://discuss.sonarsource.com/t/fp-in-ts-sonarcloud-analysis-detecting-nested-template-literal-for-styled-emotion-components/9574/4

From Greg: 

> And indeed this issue is quite painful when working with styled components. It’s quite common to nest template literal in this case.

**Comment by yassin-kammoun-sonarsource** [comment]
> Please add also the reason for the exception. Originally requested in https://discuss.sonarsource.com/t/fp-in-ts-sonarcloud-analysis-detecting-nested-template-literal-for-styled-emotion-components/9574/4
> 
> From Greg:
> 
> > And indeed this issue is quite painful when working with styled components. It’s quite common to nest template literal in this case.

I preferred to provide a general explanation of why it's okay to have nested templates spanning multiple lines rather than a specific one about styled emotion components.

**Pull request merged**